tutoroars/templates/oars/apps/clickhouse/migrations/alembic/versions/50abf505bfe6_create_xapi_events_all_table.py

```diff
@@ -0,0 +1,49 @@
+"""create xapi_events_all table
+
+Revision ID: 50abf505bfe6
+Revises: 2cb2b847d3fb
+Create Date: 2023-06-22 22:56:24.567649
+
+"""
+from alembic import op
+import clickhouse_sqlalchemy
+from clickhouse_sqlalchemy import engines
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "50abf505bfe6"
+down_revision = "2cb2b847d3fb"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "xapi_events_all",
+        sa.Column(
+            "emission_time",
+            clickhouse_sqlalchemy.types.common.DateTime(),
+            nullable=False,
+        ),
+        sa.Column(
+            "event_id", clickhouse_sqlalchemy.types.common.UUID(), nullable=False
+        ),
+        sa.Column(
+            "event_str", clickhouse_sqlalchemy.types.common.String(), nullable=False
+        ),
+        sa.PrimaryKeyConstraint("emission_time", "event_id"),
+        engines.MergeTree(
+            partition_by=sa.text("toYYYYMM(emission_time)"),
+            order_by=("emission_time", "event_id"),
+        ),
```

Comment:
@bmtcril  Alembic did not automatically generate this engine, I did it manually and formatted the file. But seems like other engines are well supported:

https://github.com/xzkostyan/clickhouse-sqlalchemy-alembic-example/blob/main/alembic/versions/ebd7d2403638_init.py#L29

Reply:
Nice!